service/substore/handlers_adminapi.go

```diff
@@ -0,0 +1,156 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018-2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package substore
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/CanonicalLtd/serial-vault/datastore"
+	"github.com/CanonicalLtd/serial-vault/service/response"
+	"github.com/gorilla/mux"
+)
+
+// APIListHandler is the API method to fetch the sub-store models
+func APIListHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	// Get the user and API key from the header
+	username := r.Header.Get("user")
+	apiKey := r.Header.Get("api-key")
```

Comment:
Maybe there could be a helper method with this code that is repeated across these different handlers:

    // Get the user and API key from the header
	username := r.Header.Get("user")
	apiKey := r.Header.Get("api-key")

	// Find the user by API key
	user, err := datastore.Environ.DB.GetUserByAPIKey(apiKey, username)
	if err != nil {
		response.FormatStandardResponse(false, "error-auth", "", err.Error(), w)
		return
	}